standards/chronologic-piper.md

```diff
@@ -0,0 +1,124 @@
+# Specification
+
+## Summary
+A standardized interface for interacting with subscriptions.
+
+##  Abstract
+This standard defines a contract interface for interacting with subscription data. The scope of this proposal is end-user / wallet interactions. Creation of the subscriptions is out-of-scope of this ERC.
+
+## Our Thoughts
+From the subscriber's side:
+
+* I want strong guarantees on when I can cancel my subscription.
+* I will normally want payments to happen automatically without any action on my part.
+* In some cases it might be valuable to require an approval process.
+* For dynamically priced subscriptions I want to be able to set limits (require authorization if subscription is more than X).
+
+
+From the providers's side:
+
+* I need the ability to charge a fixed fee per subscription time unit (netflix, pandora, etc).
+* I need the ability to charge a dynamic fee per subscription time unit (aws, twilio, etc).
+* I need to be able to create reasonably accurate forecasts for upcoming subscriptions: Programatic checks that subscription accounts have available balance and that subscription is active.
+
+## Motivation
+Standard that covers all usecases of subscription based model with minimal interface suitable for wallets.
+
+### Interface
+
+#### cancel
+
+Immediately cancels the subscription. Implementations should ensure that any unpaid subscription payments are paid to the provider as part of cancellation to ensure providers are able to let subscriptions fees fill up for arbitrary lengths of time, allowing them to reduce overhead from transaction costs.
+
+```SOLIDITY
+
+/** @dev Cancels the subscription
+  * @return true if cancellation succedded 
+*/
+
+function cancel()
```

Comment:
Standard interface should not be limited to a "contract-per-subscription" model. I believe UUID should be provided at the least.

Reply:
This is more of the architectural choice to keep contract-per-subscription model. Given that subscription is going to hold the value, regardless whether ETH or tokens, we are striving to eliminate the honeypot problem.